vivarium/framework/state_machine.py

```diff
@@ -339,11 +336,27 @@ def _normalize_probabilities(self, outputs, probabilities):
             outputs.append('null_transition')
         return outputs, probabilities/(np.sum(probabilities, axis=1)[:, np.newaxis])
 
+    def append(self, transition):
+        if not isinstance(transition, Transition):
+            raise TypeError(
+                'TransitionSet must contain only Transition objects. Check constructor arguments: {}'.format(self))
+        self.transitions.append(transition)
+
+    def extend(self, transitions):
+        for transition in transitions:
+            self.append(transition)
+
+    def __iter__(self):
+        return iter(self.transitions)
+
+    def __len__(self):
+        return len(self.transitions)
+
     def __str__(self):
-        return repr(self)
+        return repr(self.transitions)
```

Comment:
Think you still want repr(self) here so that the transitions get passed through str.

Reply:
I've changed this from being a list subclass to storing the transitions on the attribute self.transitions. So I think this does what you think it should, unless you're suggesting that there is something in the TransitionSet itself that should be captured in the string.

Reply:
Oh, no, it's just that it feels inconsistent that `str(transition_set)` essentially returns `[repr(x) for x in transition_set]` while `repr(transition_set)` returns `[str(x) for x in transition_set]`.